### ISSUE TITLE
Save editor on focus lost

### ISSUE THREAD

**Issue by guw** [eclipse-platform/eclipse.platform.ui#171]
It would be great if there is an option in Eclipse to turn on auto-save when the IDE window looses focus.

**Comment by jukzi** [comment]
I don't want such enabled as default. It can cause real trouble when saving during a build or team operation. 
However if autosave is enabled anyway it could be default to have "focus lost" as trigger. But not more often then the autosave interval.
![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/51790620/1c2e0d51-7d75-4b5d-883a-40fe2960fcc1)
@guw do you plan to contribute something or do we close it as not-planned?

**Comment by elsazac** [comment]
>However if autosave is enabled anyway it could be default to have “focus lost” as trigger. But not more often than the autosave interval.

@jukzi - just curious, if the autosave is enabled, it will anyways trigger the save after the autosave interval right? then what added benefit the `focus lost` trigger will provide?

**Comment by mickaelistria** [comment]
I'm curious about the bigger story: why does one use auto-save? Is it to prevent from data loss?
I've seen some editors doing things smarter, by keeping internally in their disk area (eg ~/eclipse-workspace/.metadata) a copy of the document that are under edition, with the state as we see it in the editor. And if the editor happens to crash, then the editor tries to restore its content from the working copy cache. This is much less intrusive than auto-save.

**Comment by jukzi** [comment]
I agree with @mickaelistria , for example notepad++ does a great job at this: it does not even ask if unchanged files should be saved on shutdown.  It just already saved it somewhere and keeps bothersome questions away from me if i have to restart my computer.

**Comment by laeubi** [comment]
Regarding autosave on focus lost, what I often encounter is the following:

1. Have two classes `A` + `B` where `A` reference `B` and their editors open
2. No in `A` type a new method that do not exits in `B`
3. Now JDT offers a quickfix to add the missing method and if I accept this is jumping to `B` where I can edit the method
4. If one now switches back to `A` it still complains that the method is missing and offers to add the method unless one saves `B`

So saving is sometimes required to let other things happen so **not** saving is most of the time not an option anyways...
